src/tools/idlc/src/descriptor.c

```diff
@@ -127,22 +64,29 @@ max_alignment(const struct alignment *a, const struct alignment *b)
   return b->ordering > a->ordering ? b : a;
 }
 
+static bool
+is_node_mutable(const idl_node_t *node)
+{
+  return (idl_is_struct(node) && ((idl_struct_t *)node)->extensibility == IDL_EXTENSIBILITY_MUTABLE) ||
+    (idl_is_union(node) && ((idl_union_t *)node)->extensibility == IDL_EXTENSIBILITY_MUTABLE);
+}
+
```

Comment:
Better make this `idl_is_mutable`? Perhaps do something like `idl_is_extensible(<node>, idl_extensibility_t)`?

Reply:
Good suggestion, fixed in 5540113